src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -2121,15 +2117,16 @@ private boolean isChangesDetected(@NonNull SCMRevision revision, @NonNull P proj
             return revisionActions;
         }
 
-        private boolean isHeadActionsFetched(@NonNull SCMHead head, @NonNull Branch branch, String rawName) {
-            boolean headActionsFetched = false;
+        private void setBranchActions(@NonNull SCMHead head, @NonNull Branch branch, @CheckForNull Branch origBranch) {
             try {
                 branch.setActions(source.fetchActions(head, event, listener));
-                headActionsFetched = true;
```

Comment:
The only reason `headActionsFetched`  existed was so that we could try an alternative later.  
Go ahead and do it in place.